File: GuideObjC/GuideObjC/ViewControllers/MoreController/MoreController.m
//
//  MoreController.m
//  GuideObjC
//
//  Created by LuzanovRoman on 12.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "MoreController.h"
#import "ProgressController.h"

@interface MoreController ()

@end

@implementation MoreController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tableView.tableHeaderView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, self.tableView.bounds.size.width, 0.1)];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell * cell = [tableView cellForRowAtIndexPath:indexPath];
    
    if ([cell.reuseIdentifier isEqual:@"LogoutCell"]) {
        [ProgressController show];
        [self.graph logoutWithCompletion:^(id object, NSError * error) {
            [ProgressController hide];
            [self.tabBarController dismissViewControllerAnimated:true completion:nil];
        }];
    }
}
@end

File: GuideSwift/GuideSwift/ViewControllers/MoreController/MoreController.swift
//
//  MoreController.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 12.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

class MoreController: UITableViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        tableView.tableHeaderView = UIView(frame: CGRect(x: 0, y: 0, width: tableView.bounds.size.width, height: 0.1))
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        guard let identifier = tableView.cellForRow(at: indexPath)?.reuseIdentifier, identifier == "LogoutCell" else { return }
        
        ProgressController.show()
        Graph.logout { (_, error) in
            ProgressController.hide()
            
            SimpleFileManager.shared.deleteAllFiles()
            self.tabBarController?.dismiss(animated: true, completion: nil)
        }
    }
}
